pkg/client/clientcmd/client_config.go

```diff
@@ -250,6 +250,9 @@ func (config DirectClientConfig) Namespace() (string, error) {
 	if len(configContext.Namespace) == 0 {
 		return api.NamespaceDefault, nil
 	}
+	if configContext.Namespace == "*" {
```

Comment:
+1, I agree with not storing the string here.  Perhaps we can add a boolean 'selectAll' which is used if the namespace is empty?